pulp-glue/pulp_glue/common/openapi.py

```diff
@@ -53,36 +55,79 @@ class AuthProviderBase:
     Returned auth objects need to be compatible with `requests.auth.AuthBase`.
     """
 
-    def basic_auth(self) -> t.Optional[t.Union[t.Tuple[str, str], requests.auth.AuthBase]]:
-        """Implement this to provide means of http basic auth."""
+    auth_type: str = ""
+
+    def auth(self, request: requests.PreparedRequest) -> requests.auth.AuthBase:
+        """Implement this to provide means of http auth."""
         return None
 
-    def __call__(
-        self,
-        security: t.List[t.Dict[str, t.List[str]]],
-        security_schemes: t.Dict[str, t.Dict[str, t.Any]],
-    ) -> t.Optional[t.Union[t.Tuple[str, str], requests.auth.AuthBase]]:
-        for proposal in security:
-            if [security_schemes[name] for name in proposal] == [
-                {"type": "http", "scheme": "basic"}
-            ]:
-                result = self.basic_auth()
-                if result:
-                    return result
-        raise OpenAPIError(_("No suitable auth scheme found."))
+    def __call__(self, request: requests.PreparedRequest) -> requests.PreparedRequest:
```

Comment:
This is an Auth provider. It's supposed to provide appropriate AuthBase objects. It's not authenticating anything.

Reply:
It's returning an `AuthBase` compatible object, like stated [here](https://github.com/pulp/pulp-cli/blob/375f8b0952e33c409eed58c396fe0feedb7e43eb/pulp-glue/pulp_glue/common/openapi.py#L55) and based on [this](https://requests.readthedocs.io/en/latest/_modules/requests/auth/#AuthBase).
If this is about the name of the `auth` method it was based on the previous one `basic_auth`, which I think it's related only to HTTP Basic Authentication method.

Reply:
It is supposed to _return_, not _be_ an AuthBase.